nucypher/config/base.py

```diff
@@ -220,11 +229,17 @@ def _write_configuration_file(self, filepath: str, override: bool = False) -> st
 
     def serialize(self, serializer=json.dumps) -> str:
         """Returns the JSON serialized output of `static_payload`"""
-        serialized_payload = serializer(self.static_payload(), indent=self.INDENTATION)
+        payload = self.static_payload()
+        payload['version'] = self.VERSION
+        serialized_payload = serializer(payload, indent=self.INDENTATION)
         return serialized_payload
 
     @classmethod
     def deserialize(cls, payload: str, deserializer=json.loads) -> dict:
         """Returns the JSON deserialized content of `payload`"""
         deserialized_payload = deserializer(payload)
+        version = deserialized_payload.pop('version', UNKNOWN_VERSION)
+        if version != cls.VERSION:
```

Comment:
Not sure I fully understand - are we not backwards compatible with older versions of the config file? Is that something we want?

Reply:
Yes that's the intention - If we modify the format or scheme of this config file, new versions of the parser will backwards-incompatible and this flag helps indicate that intentionally instead of with JSON decode handling.

Reply:
So just to clarify, the `OldVersion` error would be explicitly caught, thereby signalling an older version of the config, and some logic used to still be able to parse it - thereby allowing backwards compatibility? And the code that would do this is not yet present/not yet needed since this is the initial version?

Reply:
I do not think there's any reason to provide back compatibility for configuration serialization -Simply recreating or editing the JSON  works as an "update". 

Consider: We dont exactly know what sceme changes may be introduced in the future. If someone updates nucypher, and the new version uses a new JSON configuration scheme, but still has and older versioned configuration file on the disk, what shall we do?

Here are some potential options:

A - Always keep (never update) the previous parser scheme, ensure back compatibility by hardcoding the versions in the codebase.  This will end up looking like: `class CharacterConfigurationV1` `class CharacterConfigurationV2`

B - Use a flag on the class (`CharacterConfiguration.VERSION`). Create a migrations python/bash script that is automatically triggered everytime this exception is caught to make the specific edits to the file needed.

C - Use a flag on the class (`CharacterConfiguration.VERSION`). Verify the version it the latest and crash if it isn't, informing the user they need to recreate/edit their configuration.

Reply:
It's a fair point that A & B are rabbit holes: 
* A = code debt for versioned classes
* B = code debt for scripts and potential situations of multiple version jumps in one step.

I'm with you that C makes the most sense